Allow the use of github auth tokens.

jcvernaleo opened the pull request:
This allows users with two factor auth enabled to use pull-report.

This closes #6.

ryan-roemer commented on README.md:
```diff
@@ -32,6 +32,7 @@ $ pull-report --help
     --html                Display report as HTML
     --gh-user <username>  GitHub user name
     --gh-pass <password>  GitHub pass
+	--gh-token <token>    Github token
```
Alignment (probably need spaces, not tabs)

ryan-roemer commented on pull-report.js:
```diff
@@ -171,9 +172,12 @@ if (require.main === module) {
-      "on the command line");
+  // If we have a token, no need for user/password
+  if (!program.ghToken) {
+    if (!(program.ghUser && program.ghPass)) {
```
Combine to a single conditional please.

ryan-roemer commented on pull-report.js:
```diff
@@ -171,9 +172,12 @@ if (require.main === module) {
+  if (!program.ghToken) {
+    if (!(program.ghUser && program.ghPass)) {
+      throw new Error("Must specify GitHub user / pass in .gitconfig or " +
+		      "on the command line");
```
Trailing lines get indented at two per project style (no worries -- not documented :wink: )

ryan-roemer commented on pull-report.js:
```diff
@@ -229,11 +233,21 @@ if (require.main === module) {
+      username: program.ghUser,
+      password: program.ghPass,
+    });
+  }else{
```
NIT: `} else {`

ryan-roemer commented:
@jcvernaleo -- Thanks for the PR! Have some nits and refactors...

jcvernaleo commented:
Sorry about those.  I've been spoiled by using golang (where there is a tool to enforce all those style issues that everyone uses so I don't always remember to pay attention to them).  I think that addresses all the points.

ryan-roemer commented on README.md:
```diff
@@ -32,6 +32,7 @@ $ pull-report --help
     --html                Display report as HTML
     --gh-user <username>  GitHub user name
     --gh-pass <password>  GitHub pass
+    -gh-token <token>    Github token
```
NIT: `--gh-token` (only has one dash right now).

ryan-roemer commented on pull-report.js:
```diff
@@ -171,9 +172,10 @@ if (require.main === module) {
+  // If we have a token, no need for user/password
+  if (!program.ghToken && !(program.ghUser && program.ghPass)) {
+      throw new Error("Must specify GitHub user / pass in .gitconfig or " +
+	"on the command line");
```
NIT: Just one more space on the indent here.

ryan-roemer commented:
@jcvernaleo -- Two nits and after that I'll try and do a release tomorrow. Thanks!

jcvernaleo commented:
Sorry about that.  This time I checked it even more carefully against master so it should match up.

ryan-roemer commented:
Released in `pull-report@0.1.0`
